lib/sprangular/order_updater.rb

```diff
@@ -98,6 +98,11 @@ def update_order
     end
 
     def find_selected_rate(order)
+      # When changing zones, don't persist a shipment 
+      # that has been or will be deleted as part of the callback chain
+      if order.state == 'address' || order.state == 'delivery'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.